charts/consul/templates/gateway-resources-configmap.yaml

```diff
@@ -26,14 +26,13 @@ data:
         kind: gatewayClassConfig
         spec:
           deployment:
-            resources:
-              {{ .Values.meshGateway.resources }}
+            resources: {{ toJson .Values.meshGateway.resources }}
```

Comment:
this needs to be converted to json, the yaml parser fails because this value without the `toJson` ends up being a string representation of a go map, the `toYaml` function ends up throwing some errors because of how it represents the map